packages/client/src/generation/utils/buildDirname.ts

```diff
@@ -34,12 +35,23 @@ function buildDirnameFind(defaultRelativeOutdir: string, runtimePath: string) {
 
   return `
 const { findSync } = require('${runtimePath}')
+const fs = require('fs')
 
-const dirname = findSync(process.cwd(), [
+// some frameworks or bundlers replace or totally remove __dirname
+const hasDirname = typeof __dirname !== 'undefined' && __dirname !== '/'
+
+// will work in most cases, ie. if the client has not been bundled
+const regularDirname = hasDirname && fs.existsSync(path.join(__dirname, 'schema.prisma')) && __dirname
+
+// if the client has been bundled, we need to look for the folders
+const foundDirname = !regularDirname && findSync(process.cwd(), [
     ${defaultRelativeOutdir ? `${JSON.stringify(defaultRelativeOutdir)},` : ''}
     ${serverlessRelativeOutdir ? `${JSON.stringify(serverlessRelativeOutdir)},` : ''}
-], ['d'], ['d'], 1)[0] || __dirname`
+], ['d'], ['d'], 1)[0]
+
+const dirname = regularDirname || foundDirname || __dirname`
```

Comment:
Btw, does it make sense to have `__dirname` as a final fallback here? If `regularDirname` is false, then `__dirname` is either undefined or doesn't have a schema, so it might be more useful to just throw an error at this point.

Reply:
Oh, I see. Makes sense.